src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -99,11 +100,11 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
                 EntityRef owner = item.getOwner();
                 final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
                 //TODO: replace with better fluid handling, maybe by new CoreFluids module
+                FluidContainerItemComponent removedFluidContainer= removedItem.getComponent(FluidContainerItemComponent.class);
+                float volume= Math.min((removedFluidContainer.volume + removedFluidContainer.fillingAmount), removedFluidContainer.maxVolume);
```

Comment:
When a fluid container items does not have an entry for `fillingAmount` the value will default to `0.0` - and filling the container won't work. I think we should add a check whether the amount is smaller or equal to 0 and set it to `maxVolume` in that case.